### DISCUSSION TITLE
How to use forecast_overlap properly in the handoff_forecast lstm

### DISCUSSION THREAD

**Discussion by justagoodguy-44** [neuralhydrology/neuralhydrology#178] [Q&A]
Hi, I'm trying to use the handoff_forecast lstm, and I'm a bit unsure about how to use the forecast_overlap property correctly. 

My first confusion is that in the docs it is said that: "The handoff forecast LSTM model can implement a delayed handoff as well, such that the handoff between the hindcast and forecast LSTM occurs prior to the forecast issue time". But when I look in the code it seems that the overlap actually extends the hindcast lstm's predictions into the forecast period?

I'm also unsure what is happening when both lstms are making their predictions simultaneously. Is this only used in the regularization parameter when we add forecast_overlap as a regularizer?

Finally on a more practical side I'm just not sure how to use this param effectively. I have a 365 day hindcast as 112 day forecast, and for example if I try to set the overlap to 50 I get rather poor performance for predictions 0 to 50, then it goes up (as shown in the attached image).

![Screenshot 2024-04-26 152046](https://github.com/neuralhydrology/neuralhydrology/assets/61226987/b56bd57a-3f3d-47ac-94ee-960a75c75791)

Sorry for the long post, and I wish I could just read up more on the theory of this instead of having to bother someone but I've not been able to find source material that explains this concept, so if you have a reference I'd be more than happy to read up on it.

Thanks a lot

**Comment by brett-snider** [comment]
Hi @justagoodguy-44, I just came across your post here and I'm running into similar issues/questions. 

Specifically I was wondering how you came up with that graph of NSE vs forecast timestep. Have you set your predict_last_n to 112? or do you just have it as predict_last_n = 1? When I set predict_last_n = 1, I only get one prediction. 

Any help would be greatly appreciated.
